tests/unit/mixins/route-test.js

```diff
@@ -402,37 +402,63 @@ test('it allows overrides/manual invocations of updateInfinityModel', assert =>
 });
 
 test('It allows to set startingPage as 0', assert => {
-  var RouteObject = Ember.Route.extend(RouteMixin, {
-    model() {
-      return this.infinityModel('item', {startingPage: 0});
+  var route = createRoute('item', {
+    store: createDummyStore({
+      items: [{id: 1, name: 'Test'}],
+      meta: {
+        total_pages: 1
+      }
+    })
+  }, {startingPage: 0});
+
+  captureModel(route);
+
+  assert.equal(0, route.get('currentPage'));
+  assert.equal(true, route.get('_canLoadMore'));
+});
+
+test('it calls the afterInfinityModel method on objects fetched from the store', assert => {
+  var route = createRoute('item', {
+    store: {
+      query() {
+        var item = { id: 1, title: 'The Great Gatsby' };
+        return new Ember.RSVP.Promise(resolve => {
+          Ember.run(this, resolve, Ember.ArrayProxy.create({
+            content: Ember.A([item])
+          }));
+        });
+      }
+    },
+    afterInfinityModel(items) {
+      return items.setEach('author', 'F. Scott Fitzgerald');
     }
   });
-  var route = RouteObject.create();
 
-  var dummyStore = {
-    query() {
-      return new Ember.RSVP.Promise(resolve => {
-        Ember.run(this, resolve, Ember.Object.create({
-          items: [{id: 1, name: 'Test'}],
-          meta: {
-            total_pages: 1
-          }
-        }));
-      });
-    }
-  };
+  var model = captureModel(route);
 
-  route.set('store', dummyStore);
+  assert.equal(model.get('content.firstObject.author'), 'F. Scott Fitzgerald', 'updates made in afterInfinityModel should take effect');
+});
 
-  var model;
-  Ember.run(() => {
-    route.model().then(result => {
-      model = result;
-    });
+test('it resolves promises returned by the afterInfinityModel method', assert => {
+  var route = createRoute('item', {
+    store: {
+      query() {
+        var item = { id: 1, title: 'The Great Gatsby' };
+        return new Ember.RSVP.Promise(resolve => {
+          Ember.run(this, resolve, Ember.ArrayProxy.create({
+            content: Ember.A([item])
+          }));
+        });
+      }
+    },
+    afterInfinityModel(items) {
+      return items.setEach('author', 'F. Scott Fitzgerald');
```

Comment:
Hm, this test doesn't actually test that - in fact it's identical to the test above.